### PR TITLE
feat(dop): performance measure metrics

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
performance measure metrics

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  Support performance measure metrics            |
| 🇨🇳 中文    |   实现效能度量指标           |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: apistructs/gittar.go [hunk 1]
```diff
@@ -837,3 +837,10 @@ type MergeBaseResponse struct {
 	Header
 	Data *Commit `json:"data"`
 }
+
+type GittarListRepoRequest struct {
+	OrgIDS     []uint64   `json:"orgIDS"`
```
Comment (sfwn) [inline apistructs/gittar.go:842]: IDs

File: apistructs/issue.go [hunk 2]
```diff
@@ -1159,13 +1159,16 @@ type IssueTestCaseRelationsListRequest struct {
...
 type IssuesStageRequest struct {
-	StatisticRange string  `json:"statisticRange"` //事件类型 项目/迭代
-	RangeID        int64   `json:"rangeId"`        //项目id/迭代id
-	StateIDS       []int64 `json:"StateIDs"`       // state id list
+	Assignee       uint64    `json:"assignee"`
+	Owner          uint64    `json:"owner"`
+	IssueType      IssueType `json:"issueType"`
+	StatisticRange string    `json:"statisticRange"` //事件类型 项目/迭代
+	RangeID        int64     `json:"rangeId"`        //项目id/迭代id
+	StateIDS       []int64   `json:"StateIDs"`       // state id list
```
Comment (sfwn) [inline apistructs/issue.go:1167]: IDs

File: cmd/erda-server/bootstrap.yaml [hunk 3]
```diff
@@ -288,6 +288,7 @@ dop: { }
 erda.core.dop.taskerror: { }
 erda.dop.qa.unittest: { }
 project-management-report: { }
+performance-measure: { }
```
Comment (sfwn) [inline cmd/erda-server/bootstrap.yaml:291]: efficiency-measurement

File: internal/apps/dop/providers/performance_measure/cache.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+func genReopenStateIDSKey(projectID uint64) string {
+	return fmt.Sprintf("reopen_state_ids_%d", projectID)
+}
+
+func genResolvedStateIDSKey(projectID uint64) string {
+	return fmt.Sprintf("resolved_state_ids_%d", projectID)
+}
+
+func (p *propertyCache) SetWontfixStateIDS(projectID uint64, stateIDS []uint64) {
```
Comment (sfwn) [inline internal/apps/dop/providers/performance_measure/cache.go:80]: IDs

File: internal/tools/gittar/metrics/collector.go [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+		errors: prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: "gittar",
+			Name:      "scrape_error",
+			Help:      "1 if there was an error while getting personal contributor metrics, 0 otherwise",
+		}),
+		svc: svc,
+	}
+}
+
+func (c *Collector) RefreshPersonalContributors() error {
```
Comment (sfwn) [inline internal/tools/gittar/metrics/collector.go:119]: PersonalContributions

File: internal/core/openapi/legacy/api/apis/dop/personal_contributor-query.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dop
+
+import "github.com/erda-project/erda/internal/core/openapi/legacy/api/apis"
+
+var PersonalContributorQuery = apis.ApiSpec{
+	Path:          "/api/personal-contributor/actions/query",
```
Comment (sfwn) [inline internal/core/openapi/legacy/api/apis/dop/personal_contributor-query.go:20]: contribution

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by sfwn** [comment]
How to fill in historical commit data?

File: cmd/erda-server/bootstrap.yaml (hunk 3 above)
Comment (chengjoey) [reply cmd/erda-server/bootstrap.yaml]: done, and adujsted other `performance` title name

File: internal/apps/dop/providers/performance_measure/cache.go (hunk 4 above)
Comment (chengjoey) [reply internal/apps/dop/providers/performance_measure/cache.go]: done

File: internal/tools/gittar/metrics/collector.go (hunk 5 above)
Comment (chengjoey) [reply internal/tools/gittar/metrics/collector.go]: done

File: internal/core/openapi/legacy/api/apis/dop/personal_contributor-query.go (hunk 6 above)
Comment (chengjoey) [reply internal/core/openapi/legacy/api/apis/dop/personal_contributor-query.go]: done

**Comment by chengjoey** [comment]
> How to fill in historical commit data?

third-party applications request gittar `/personal-contribution` and then convert it into metrics

File: cmd/erda-server/conf/i18n/cp/scenarios/personal-workbench.yaml [hunk 7]
```diff
@@ -2,6 +2,7 @@ zh:
   Count: 数
   project: 项目
   app: 应用
+  performance-measure: 效能度量
```
Comment (sfwn) [inline cmd/erda-server/conf/i18n/cp/scenarios/personal-workbench.yaml:5]: Adjust keys in another pr please.

**Review by sfwn — APPROVED** [review]

**Pull request merged**